File: biztrack/biztrack/report/application_tracking_report/application_tracking_report.js
// Copyright (c) 2025, Shivanand and contributors
// For license information, please see license.txt

frappe.query_reports["Application Tracking Report"] = {
    use_container_width: false,
    scroll_x: true,
    
    filters: [
        {
            fieldname: "employee",
            label: __("Employee Email"),
            fieldtype: "Link",
            options: "User",
        },
        {
            fieldname: "employee_name",
            label: __("Employee Name"),
            fieldtype: "Data",
            width: "100px"
        },
        {
            fieldname: "event_name",
            label: __("Event Name"),
            fieldtype: "Data",
            width: "100px"
        },
        {
            fieldname: "from_date",
            label: __("From Date"),
            fieldtype: "Date",
            default: frappe.datetime.add_months(frappe.datetime.get_today(), -1),
            width: "100px"
        },
        {
            fieldname: "to_date",
            label: __("To Date"),
            fieldtype: "Date",
            default: frappe.datetime.get_today(),
            width: "100px"
        }
    ],
    
    "formatter": function(value, row, column, data, default_formatter) {
        value = default_formatter(value, row, column, data);
        
        // Format the Applications column to show as a badge if it's a number
        if (column.fieldname == "applications" && data && data.applications) {
            if (!isNaN(data.applications)) {
                let count = parseInt(data.applications);
                let color = count > 10 ? "red" : count > 5 ? "orange" : "green";
                value = `<span class="badge badge-${color}">${count}</span>`;
            }
        }
        
        // Format the duration column
        if (column.fieldname == "duration" && data && data.duration) {
            value = `<span style="color: #4CAF50; font-weight: bold;">${data.duration}</span>`;
        }
        
        // Format the new Name column (name1) with custom styling
        if (column.fieldname == "name1" && data && data.name1) {
            value = `<span style="color: #2196F3; font-weight: 500;">${data.name1}</span>`;
        }
        
        // Format the Application ID column
        if (column.fieldname == "application_id" && data && data.application_id) {
            value = `<span style="color: #FF9800; font-weight: 500;">${data.application_id}</span>`;
        }
        
        // Format the Event ID column
        if (column.fieldname == "event_id" && data && data.event_id) {
            value = `<span style="color: #9C27B0; font-weight: 500;">${data.event_id}</span>`;
        }
        
        // Format the Idle Duration column
        if (column.fieldname == "idle_duration" && data && data.idle_duration) {
            let duration = parseFloat(data.idle_duration);
            if (!isNaN(duration)) {
                // Format as decimal with 2 places and add styling
                value = `<span style="color: #FF5722; font-weight: 500;">${duration.toFixed(2)}</span>`;
            }
        }
        
        return value;
    },
    
    "onload": function(report) {
        // Add custom button to export data
        report.page.add_inner_button(__("Export to Excel"), function() {
            frappe.utils.csvify(report.data, report.columns, "Applications Tracking Report");
        });
        
        // Add refresh button
        report.page.add_inner_button(__("Refresh"), function() {
            report.refresh();
        });
    },
    
    "initial_depth": 0,
    "tree": false,
    "name_field": "name",
    "parent_field": "",
    "disable_prepared_report": false
};

File: biztrack/biztrack/report/application_tracking_report/application_tracking_report.py
# Copyright (c) 2025, Shivanand and contributors
# For license information, please see license.txt

# import frappe

import frappe

def execute(filters=None):
    """
    Applications Tracking Report
    Shows all draft Applications Tracking records with complete details
    """
    
    # Build filter conditions
    conditions = {"docstatus": 0}
    
    # Add additional filters if provided
    if filters:
        if filters.get("employee"):
            conditions["employee"] = filters.get("employee")
        if filters.get("employee_name"):
            conditions["employee_name"] = ["like", f"%{filters.get('employee_name')}%"]
        if filters.get("event_name"):
            conditions["event_name"] = ["like", f"%{filters.get('event_name')}%"]
        if filters.get("from_date"):
            conditions["creation"] = [">=", filters.get("from_date")]
        if filters.get("to_date"):
            if conditions.get("creation"):
                conditions["creation"] = ["between", [filters.get("from_date"), filters.get("to_date")]]
            else:
                conditions["creation"] = ["<=", filters.get("to_date")]
    
    # Fetch data from Applications Tracking doctype
    data = frappe.db.get_all(
        "Applications Tracking",
        filters=conditions,
        fields=[
            "name",
            "employee_name", 
            "employee",
            "event_name",
            "duration",
            "applications",
            "name1",           # New field added
            "application_id",  # New field added
            "event_id",        # New field added
            "idle_duration"    # New field added
        ],
        order_by="creation desc"
    )
    
    # Define columns with exact specifications from your SQL query
    columns = [
        {
            "label": "ID",
            "fieldname": "name",
            "fieldtype": "Link",
            "options": "Applications Tracking",
            "width": 120
        },
        {
            "label": "Employee Name",
            "fieldname": "employee_name",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": "Employee Email",
            "fieldname": "employee",
            "fieldtype": "Link",
            "options": "User",
            "width": 120
        },
        {
            "label": "Event ID",          # New column added
            "fieldname": "event_id",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": "Event Name",
            "fieldname": "event_name",
            "fieldtype": "Data",
            "width": 150
        },
        {
            "label": "Application ID",    # New column added
            "fieldname": "application_id",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": "Applications",
            "fieldname": "applications",
            "fieldtype": "Data",
            "width": 150
        },
        {
            "label": "Spent Time",
            "fieldname": "duration",
            "fieldtype": "Data",
            "width": 150
        },
        {
            "label": "Name",              # New column added
            "fieldname": "name1",
            "fieldtype": "Data",
            "width": 120
        },
        {
            "label": "Idle Duration",     # New column added
            "fieldname": "idle_duration",
            "fieldtype": "Float",
            "width": 120
        }
    ]
    
    return columns, data